lfs/pointer.go

```diff
@@ -95,8 +95,16 @@ func DecodePointerFromFile(file string) (*Pointer, error) {
 		return nil, err
 	}
 	defer f.Close()
-	return DecodePointer(f)
+	
+	// Since the lower level functions don't have any awareness of which file caused the error
+	// log the filename here when the error eventually bubbles up
+	p, err := DecodePointer(f)
+	if err != nil {
+		errors.NewNotAPointerError(errors.New("expected file to contain a pointer, but it didn't. File: " + file))
```

Comment:
I think this fails to pass on the message from lower down.  It can be useful for folks to know what exactly made it invalid instead of having to guess.  Also, this doesn't actually change anything, since it just instantiates an object here and doesn't return anything. 

Maybe something like this:

```go
if err != nil {
        err = errors.NewNotAPointerError(fmt.Errorf("expected %q to contain a pointer, but it didn't: %s", file, err))
}
```

And of course, some tests, so we can have confidence that this does work as expected.  There are tests in `t` that would be a good place for this.

Reply:
Passing on the error is a good idea, I'll fix that. 

I'll try taking a look at test as well. Found a very similar test in t-smudge.sh called "smudge with invalid pointer". However I'm not sure if that runs the code path I'm testing, since it reads the data directly from stdin, rather than from a file.